src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -440,7 +442,7 @@ protected function build(ContainerBuilder $container)
      */
     protected function getContainerClass()
     {
-        return str_replace('\\', '_', \get_class($this)).ucfirst($this->environment).($this->debug ? 'Debug' : '').'Container';
+        return $this->name.str_replace('\\', '_', \get_class($this)).ucfirst($this->environment).($this->debug ? 'Debug' : '').'Container';
```

Comment:
Keeping the `$this->name` prefix is a requirement here. The failing tests on lower branches hint this is a BC break for multi-kernel apps that generate a different container per name.
Once ppl will have moved out of using name, then we will be free to remove it - in 5.0.